### PR TITLE
Added translations

### PR DESCRIPTION
**Pull request by boerniee**
Hello,
I have added vue-i18n for translations. The language to use will be determined by the browsers language.
Currently only english and german language is translated.

Not included is formatting dates and fixing the currency formatting.

### CODE REVIEW

File: ui/src/locales/en.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    "quickentry": "No Quick Entries | Quick Entry | Quick Entries",
+    "statistics": "Statistics",
+    "thisweek": "This week",
+    "thismonth": "This month",
+    "pastxdays": "Past one day | Past {count} days",
+    "pastxmonths": "Past one month | Past {count} months",
+    "thisyear": "This year", 
+    "alltime": "All Time",
+    "noattachments": "No Attachments so far",
+    "attachments": "Anhänge",
```
Comment (solluh) [inline ui/src/locales/en.json:11]: That's still german

**Comment by AlfHou** [comment]
Thank you for your PR :) 
There are some keys in German that should be in English, it would be great if you could double check all the keys. Also, there are some parts that are not translated to German (See attached screenshots. Might be more as well).

Lastly, it would be nice with an option in the settings to change the language. If you don't feel like doing it, I can also add it later.

image
image
image

**Comment by solluh** [comment]
I'd really love to do that.
Problem: I'm a newbee, who don't know how to make changes. Do I have to make a fork of boernies fork and do a new pull request?

**Comment by solluh** [comment]
This can't be the way, can it?

**Comment by AlfHou** [comment]
The above comment was directed to @boerniee, but it's great that you want to contribute.


If you want to make changes to someone else's PR you're correct in that you would have to fork from their fork and then  open a new PR :)

**Comment by solluh** [comment]
Sorry, *I'm too dumb for this.*
After forking boerniee's fork I was looking for the file de.json in ui/src/locals - nothing...

**Comment by AlfHou** [comment]
You probably just need to check out the proper branch. Run `git checkout -b i18n`

**Comment by boerniee** [comment]
Hi @AlfHou,
sorry I have translated the missing parts and also changed the english translations.
I have also thought about a setting for the user. But for the sake of simplicity I have started with the browsers setting. It would be nice to have user setting to change it if you would like to user another language! You can add it also.
What we also should add later on is lazy loading for the translation files.

**Comment by AlfHou** [comment]
@boerniee no worries, looks good now. Let's just get it merged, then we can iterate on it in the future. Thank you for the contribution :)

**Comment by GoNzCiD** [comment]
Is not possible to set a default language by user?

**Comment by AlfHou** [comment]
For now it uses the browser locale. I have a PR #43 where I've started working on a user setting to pick this, but it's currently a bit on hold.

**Comment by thehijacker** [comment]
I would love to do Slovenian translation. Where to start?

**Pull request merged**